### ISSUE TITLE
Incorrect header matching rules key in v2 pact files

### ISSUE THREAD

**Issue by mefellows** [pact-foundation/pact-reference#449]
Sample (modified) but discovered by the V2 specs from the [terraform client](https://github.com/pactflow/terraform-provider-pact)
```
{
  "consumer": {
    "name": "terraform-client"
  },
  "interactions": [
    {
      "description": "a request to create a system account",
      "request": {
        "method": "GET",
        "path": "/foo"
      },
      "response": {
        "matchingRules": {
          "$.header.Example": {
            "match": "type"
          }
        },
        "headers": {
          "Example": "value"
        },
        "status": 200
      }
    }
 ],
  "metadata": {
    "pactRust": {
      "ffi": "0.4.16",
      "mockserver": "1.2.5",
      "models": "1.1.19"
    },
    "pactSpecification": {
      "version": "2.0.0"
    }
  },
  "provider": {
    "name": "pactflow-application-saas"
  }
}
```

The field should be `$.headers.<key>` not `$.header.<key>` as per https://github.com/pact-foundation/pact-specification/blob/version-2/testcases/response/headers/matches%20with%20regex.json#L10.

**Comment by Saup21** [comment]
https://github.com/pact-foundation/pact-go/issues/426

**Comment by rholshausen** [comment]
This has been around for a very long time. It was fixed in Pact-JVM in 2019 https://github.com/pact-foundation/pact-jvm/issues/882

**Comment by YOU54F** [comment]
This is resolved in the now released [0.4.23](https://github.com/pact-foundation/pact-reference/releases/tag/libpact_ffi-v0.4.23) version of libpact_ffi

The following pact, was generated from https://github.com/pact-foundation/pact-workshop-go, with an update to pact-go [v2.0.8,](https://github.com/pact-foundation/pact-go/releases/tag/v2.0.8) which contains libpact_ffi 0.4.23


```
{
  "consumer": {
    "name": "GoAdminService"
  },
  "interactions": [
    {
      "description": "A request to login with user 'sally'",
      "providerState": "User is not authenticated",
      "request": {
        "matchingRules": {
          "$.path": {
            "match": "regex",
            "regex": "/user/[0-9]+"
          }
        },
        "method": "GET",
        "path": "/user/10"
      },
      "response": {
        "headers": {
          "Content-Type": "application/json",
          "X-Api-Correlation-Id": "100"
        },
        "matchingRules": {
          "$.headers['Content-Type']": {
            "match": "regex",
            "regex": "application\\/json"
          },
          "$.headers['X-Api-Correlation-Id']": {
            "match": "type"
          }
        },
        "status": 401
      }
    },
    {
      "description": "A request to login with user 'sally'",
      "providerState": "User sally does not exist",
      "request": {
        "headers": {
          "Authorization": "Bearer 2019-01-01"
        },
        "matchingRules": {
          "$.headers.Authorization": {
            "match": "type"
          },
          "$.path": {
            "match": "regex",
            "regex": "/user/[0-9]+"
          }
        },
        "method": "GET",
        "path": "/user/10"
      },
      "response": {
        "headers": {
          "Content-Type": "application/json",
          "X-Api-Correlation-Id": "100"
        },
        "matchingRules": {
          "$.headers['Content-Type']": {
            "match": "regex",
            "regex": "application\\/json"
          },
          "$.headers['X-Api-Correlation-Id']": {
            "match": "type"
          }
        },
        "status": 404
      }
    },
    {
      "description": "A request to login with user 'sally'",
      "providerState": "User sally exists",
      "request": {
        "headers": {
          "Authorization": "Bearer 2019-01-01"
        },
        "matchingRules": {
          "$.headers.Authorization": {
            "match": "type"
          },
          "$.path": {
            "match": "regex",
            "regex": "/user/[0-9]+"
          }
        },
        "method": "GET",
        "path": "/user/10"
      },
      "response": {
        "body": {
          "firstName": "Sally",
          "id": 10,
          "lastName": "McSmiley Face😀😍",
          "type": "admin",
          "username": "sally"
        },
        "headers": {
          "Content-Type": "application/json",
          "X-Api-Correlation-Id": "100"
        },
        "matchingRules": {
          "$.body.firstName": {
            "match": "type"
          },
          "$.body.id": {
            "match": "type"
          },
          "$.body.lastName": {
            "match": "type"
          },
          "$.body.type": {
            "match": "regex",
            "regex": "^(admin|user|guest)$"
          },
          "$.body.username": {
            "match": "type"
          },
          "$.headers['Content-Type']": {
            "match": "regex",
            "regex": "application\\/json"
          },
          "$.headers['X-Api-Correlation-Id']": {
            "match": "type"
          }
        },
        "status": 200
      }
    }
  ],
  "metadata": {
    "pactRust": {
      "ffi": "0.4.23",
      "mockserver": "1.2.10",
      "models": "1.2.5"
    },
    "pactSpecification": {
      "version": "2.0.0"
    }
  },
  "provider": {
    "name": "GoUserService"
  }
}
```
